sites/all/modules/_custom/NC_models/NC_models.install

```diff
@@ -0,0 +1,123 @@
+<?php
+
+function NC_models_schema(){
+	$schema = array();
+
+	$schema['folders'] = array(
+		'description' => 'Table to store details of folders created by users',
+		'fields' => array(
+			'folder_id' => array(
+				'description' => 'Unique folder id which is a combination of folder name and userid of the folder owner or his private folder',
+				'type' => 'varchar',
+				'length' => 150,
+				'not null' => TRUE,
+			),
+			'folder_name' => array(
+				'description' => 'Folder name given by the user',
+				'type' => 'varchar',
+				'length' => 150,
+				'not null' => TRUE,
+			),
+			'owner' => array(
+				'description' => 'User who has created the folder',
+				'type' => 'varchar',
+				'length' => 32,
+				'not null' => TRUE,
+			),
+			'folder_type' => array(
+				'description' => 'indicates if the folder is class/non class folder',
+				'type' => 'tinyint',
+				'mysql_type' => 'TINYINT',
+				'length' => 4,
+				'not null' => TRUE,
+			),
+			'folder_class_id' => array(
+				'description' => 'class id corresponding to the folder if it is class folder',
+				'type' => 'varchar',
+				'length' => 20,
+				'not null' => TRUE,
+			),
+			'created_time' => array(
+				'description' => 'time stamp when the assignment was given',
+				'type' => 'timestamp',
+				'mysql_type' => 'TIMESTAMP',
+				'not null' => TRUE,
+			),
+			'current_status' => array(
+				'description' => 'Indicates current status of the folder (shared, unshared, ..)',
+				'type' => 'tinyint',
+				'mysql_type' => 'TINYINT',
+				'length' => 1,
+				'not null' => TRUE,
+			),
+		),
+		'primary key' => array('folder_id'),
+		'unique keys' => array(
+			'folder_owner' => array('folder_id', 'owner'),
+			'folder_id-name' => array('folder_id', 'folder_name')
+		),
+	);
+
+	$schema['shared_members'] = array(
+		'description' => 'indicates folder shared, members who they are shared with and relationship with the folder',
+		'fields' => array(
+			'shared_id' => array(
+				'type' => 'serial',
+				'length' => 10,
+			),
+			'member_name' => array(
+				'description' => 'username of the shared user', 
+				'type' => 'varchar',
+				'length' => 32,
+				'not null' => TRUE,
+			),
+			'folder_id' => array(
+				'description' => 'Currently shared folder_id',
+				'type' => 'varchar',
+				'length' => 150,
+				'not null' => TRUE,
+			),
```

Comment:
folder id should be a foreign key in this table.

Reply:
the ideas of foreign keys is to help in cascading deletes and updates. If a particular folder id is removed, then the rows with those folder id members will removed as an effect of those.